src/dmd/expressionsem.d

```diff
@@ -5854,7 +5851,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                     auto binExp = cast(EqualExp) exp.e1;
 
                     // template args
-                    Expression comp = new StringExp(loc, Token.toString(exp.e1.op));
+                    Expression comp = new StringExp(loc, Token.toString(tok));
```

Comment:
This change is unrelated and it also may be unsafe (I haven't verified, but it's possible that exp.e1 is rewritten to something else thus changing the token). I suggest doing this in a separate PR to make rusre everything goes well; also it is good practice to separate refactoring PRs from feature development PRs

Reply:
I'll revert it. The patch was working locally and this was the only discarded change